contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/spidev_test.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+import glob
+import argparse
+import subprocess
+
+
+def runcmd(command):
+    ret = subprocess.run(
+        command,
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        encoding="utf-8",
+        timeout=1,
+    )
+    return ret
+
+
+def detect_spi_node(exit_with_notice=False):
+    spi_devices = glob.glob("/dev/spidev*")
+    if spi_devices:
+        for device in spi_devices:
+            print("device: {}".format(device))
+            print()
+    elif exit_with_notice:
+        raise SystemExit("No SPI devices found under /dev/")
+
+
+def test_spi_content_consistency(spi_path):
+    """
+    Test the consistency of the SPI communication.
+
+    Args:
+        spi_path (str): The path to the SPI device.
+
+    Raises:
+        SystemExit: If no output is reported or
+        if the content is not consistent.
+    """
+
+    # Construct the command to run the SPI test
+    cmd = "spidev_test -D {} -p 'Canonical SPI Test' -v".format(spi_path)
+    print(f"Run command: {cmd}\n")
+
+    # Run the command and capture the output
+    spi_ret = runcmd([cmd])
```

Comment:
I think we should split the cmd here by shlex

Reply:
Fiexd